src/Toolkit/Toolkit.UI.Controls/MeasureToolbar/MeasureToolbar.cs

```diff
@@ -559,144 +607,35 @@ public MapView? MapView
         private static void OnMapViewPropertyChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
         {
             var toolbar = (MeasureToolbar)d;
-            var newMapView = e.NewValue as MapView;
-            if (newMapView == null)
+            if (e.NewValue is not MapView newMapView)
             {
                 throw new ArgumentException($"{nameof(MapView)} cannot be null or empty.");
             }
 
-            var oldMapView = e.OldValue as MapView;
-            if (oldMapView != null)
+            if (e.OldValue is MapView oldMapView)
             {
-                oldMapView.SketchEditor = toolbar._originalSketchEditor;
                 oldMapView.GeoViewTapped -= toolbar.OnMapViewTapped;
                 toolbar.RemoveMeasureFeatureResultOverlay(oldMapView);
             }
 
             newMapView.GeoViewTapped += toolbar.OnMapViewTapped;
-            toolbar._originalSketchEditor = newMapView.SketchEditor;
-            toolbar.DisplayResult(newMapView.SketchEditor?.Geometry);
-        }
-
-        /// <summary>
-        /// Gets or sets the sketch editor used for measuring distances.
-        /// </summary>
-        public SketchEditor? LineSketchEditor
-        {
-            get { return GetValue(LineSketchEditorProperty) as SketchEditor; }
-            set { SetValue(LineSketchEditorProperty, value); }
+            toolbar._geometryEditor = newMapView.GeometryEditor!;
```

Comment:
Don't do this until start of editing. Clear it out again when the editor stops or mapview is null.
Having null-forgiveness operator here feels wrong (isn't _geometryEditor already nullable?).